R/add_md.R

```diff
@@ -21,13 +27,107 @@ add_nodes_to_body <- function(body, nodes, where = 0L) {
   }
 }
 
+
+#' Insert markdown before or after a set of nodes
+#'
+#' @inheritParams add_md
+#' @param md markdown text to insert
+#' @param nodes a character vector of an XPath expression OR an `xml_node` or
+#'   `xml_nodeset` object. 
+#' @param space when `TRUE` (default) inline nodes have a single space appended
+#'   or prepended to avoid the added markdown abutting text.
+#' @return a copy of the XML object with the translated markdown inserted
+#'
+#' @note The markdown content must be of the same type as the XML nodes, either
+#'   inline or block content.
+#' @dev
+insert_md <- function(body, md, nodes, where = "after", space = TRUE) {
+  new <- md_to_xml(md)
+  shove_nodes_in(body, new, nodes = nodes, where = where, space = space)
+  copy_xml(body)
+}
+
+shove_nodes_in <- function(body, new, nodes, where = "after", space = TRUE) {
+  if (inherits(nodes, "character")) {
+    xpath <- nodes
+    nodes <- xml2::xml_find_all(body, nodes, ns = md_ns())
+  } else {
+    xpath <- NULL
+  }
+  if (length(nodes) == 0) {
+    msg <- glue::glue("No nodes matched the expression {sQuote(xpath)}")
+    rlang::abort(msg, class = "insert-md-xpath")
```

Comment:
why not use cli instead?

Reply:
I keep forgetting about the fact that they have better error handlers! We don't import it, but I get the feeling that we could add cli as a dependency for this since it's pretty light.